lib/SPIRV/OCLToSPIRV.cpp

```diff
@@ -70,6 +70,23 @@ static size_t getOCLCpp11AtomicMaxNumOps(StringRef Name) {
       .Default(0);
 }
 
+static Type *getBlockStructType(Value *Parameter) {
+  // In principle, this information should be passed to us from Clang via
+  // an elementtype attribute. However, said attribute requires that the
+  // function call be an intrinsic, which it is not. Instead, we rely on being
+  // able to trace this to the declaration of a variable: OpenCL C specification
+  // section 6.12.5 should guarantee that we can do this.
+  Value *UnderlyingObject = getUnderlyingObject(Parameter);
+  Type *ParamType = nullptr;
+  if (auto *GV = dyn_cast<GlobalValue>(UnderlyingObject))
+    ParamType = GV->getValueType();
+  else if (auto *Alloca = dyn_cast<AllocaInst>(UnderlyingObject))
+    ParamType = Alloca->getAllocatedType();
+  else
+    assert(false && "Blocks in OpenCL C must be traceable to allocation site");
```

Comment:
Suggest to use `llvm_unreachable` here instead of an assert. Or the intetion is to hide the branch in release build? In this case `getBlockStructType` could return nullptr making static analyzer tools unhappy.

Reply:
Will change.